### PR TITLE
[JB] Properly overide jb config and cache folder locations

### PR DESCRIPTION
**Pull request by jeanp413**
## Description

Properly overide jb config and cache folder locations since version 2023.3.3

## Related Issue(s)

Fixes https://linear.app/gitpod/issue/EXP-1321

## How to test

- Preview env: https://jp-bizarre-quokka.preview.gitpod-dev.com/workspaces
- Test jb editors start sucessfully: 
    - you can use https://github.com/jeanp413/jenkins repo for intelllij

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

**Review by akosyakov — COMMENTED** [review]
We need to make sure that following still work
- https://www.gitpod.io/docs/references/ides-and-editors/intellij#install-plugins-from-a-workspace-image
- https://www.gitpod.io/docs/references/ides-and-editors/intellij#configure-ide-settings-per-project

We should be also conscious what happens if someone restarts a workspace which was first started with 2023.3.2 but then restarted with 2023.3.3. Would be old configs/caches picked up or completely ignored? Ignoring maybe is alright.

File: components/ide/jetbrains/launcher/main.go [hunk 1]
```diff
@@ -637,6 +642,25 @@ func handleSignal() {
...
+	buffer, err := os.ReadFile(platformOptionsPath)
+	if err != nil {
+		return err
+	}
+
+	content := string(buffer)
+
+	newContent := strings.Join([]string{
+		content,
+		fmt.Sprintf("idea.config.path=%s", configDir),
```
Comment (akosyakov) [inline components/ide/jetbrains/launcher/main.go:655]: We could instead make it a part of vm options with`-Didea.config.path` and `-Didea.system.path`. Not sure about precedence.

File: components/ide/jetbrains/launcher/main.go [hunk 2]
```diff
@@ -637,6 +642,25 @@ func handleSignal() {
 	log.Info("asked IDE to terminate")
 }
 
+func updatePlatformProperties(platformOptionsPath string, configDir string, systemDir string) error {
```
Comment (akosyakov) [inline components/ide/jetbrains/launcher/main.go:645]: this function should make sure that it does not change file unnecessary if we restart JB backend, i.e. kill lunacher process

File: components/ide/jetbrains/launcher/main.go (hunk 1 above)
Comment (jeanp413) [reply components/ide/jetbrains/launcher/main.go]: idea.properties have precedence as it's appended later https://github.com/JetBrains/intellij-community/blob/4a918183ef774c275cb12ec31e74c50c9d841ead/native/XPlatLauncher/src/lib.rs#L183-L200

**Comment by akosyakov** [comment]
@jeanp413 Can we change the integration test somehow that we catch if it happens again? i.e. add a test with an alien plugin or check that folder locations did not check on CI?

**Review by akosyakov — APPROVED** [review]
works well

File: components/ide/jetbrains/launcher/main.go [hunk 3]
```diff
@@ -637,6 +632,46 @@ func handleSignal() {
 	log.Info("asked IDE to terminate")
 }
 
+func updatePlatformProperties(platformOptionsPath string, configDir string, systemDir string) (bool, error) {
```
Comment (akosyakov) [inline components/ide/jetbrains/launcher/main.go:635]: I did not test restarting IDE, maybe make sense to write a unit test for it

**Comment by jeanp413** [comment]
Integration test pass :tada:  
image

**Pull request merged**